### ISSUE TITLE
Reserving resources

### ISSUE THREAD

**Issue by abretaud** [usegalaxy-eu/tiaas2#25]
Hi!
We're currently setting up tiaas on usegalaxy.fr (it's online on https://usegalaxy.fr/tiaas/ but not yet advertised!) and I'm wondering how we could automate the reservation of computing resources when a training is approved. We have a Slurm cluster where we could use reservations.
Is this kind of thing automated on other usegalaxy.* servers, or does it always require a manual action? (PRs to create vms maybe?)
We have sorting hat in place, so once the resources are reserved it should be easy to make use of them

**Comment by gmauro** [comment]
We are currently doing it manually with a PR, e.g. https://github.com/usegalaxy-eu/vgcn-infrastructure/pull/136
but @beatrizserrano has project to automatize it sooner or later

**Comment by abretaud** [comment]
Ok good to know! We'll let you know if we come up with something for slurm

**Comment by hexylena** [comment]
> Is this kind of thing automated on other usegalaxy.* servers

Depends on the server also. .org has a special queue with stricter compute time limits / memory usage for training and stuff gets sent there. Then there's less configuration required -- everything goes to the training queue. If you're using slurm and have those reservations that would probably be a good option for you. EU's system was developed largely because we did have some elastic cloud resources.

**Comment by abretaud** [comment]
Ok, the plan for us is to create reservations for each training, so something more dynamic than .org.
I guess each server will have its own mechanisms, if we code something, ideally it'll need to be very customizable

**Comment by hexylena** [comment]
We've added one cron job so that part is easier now to automate, let me know what you come up with, maybe we can build it into the tiaas cron jobs to have some hooks that people can script a bit.

**Comment by abretaud** [comment]
Yep, I'll let you know if/when I have something.
Slightly unrelated question: when you approve a training, there's no automatic email sent to the person who made the request?

**Comment by hexylena** [comment]
Nope. It's something that should be added, until now everyone is handling it manually

**Comment by gmauro** [comment]
Not at the moment, @beatrizserrano uses a template, specifying the detail.
